### PR TITLE
Dsp documentation

### PR DESCRIPTION
**Pull request by iliaplatone**

### CODE REVIEW

**Comment by knro** [comment]
Great! Do you have more commits pending or should I merge this?

**Comment by iliaplatone** [comment]
I guess you can merge as is right now.
Please take a look into defaultdevice: I added two methods that use the DSP API, I put them there to get them usable by any driver, I hope these are not too slow on embedded devices. Also the base type is double so in vfp machines there should be some problem.
An altrnative should be putting them into indiclient, let me know what do you think.

**Comment by iliaplatone** [comment]
... ops. sorry, wait, there's an issue on mac

**Comment by knro** [comment]
Ok, this is a lot more than documentation! I just checked all the changes:

1. DefaultDevice should not have any DSP related functions. Why can't these functions move to indidetector.cpp ?
2. astro.c and align.c can be merged in indicom

**Comment by iliaplatone** [comment]
The DSP functions can be used on CCD, for example to do some kind of sharpen or wavelets. Since is bothering reproduce these methods each time I put them there.
Is it possible to add a specific class for DSP functions to be inherited by single drivers?

**Comment by knro** [comment]
isn't the dsp lib suppose to provide such functions then?

**Comment by iliaplatone** [comment]
Yes but sometimes is tedious rewrite those functions.
I'll move them into Detector.

File: libindi/libs/indibase/basedevice.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 #include "indibase.h"
 #include "indiproperty.h"
+#include "dsp.h"
```
Comment (knro) [inline libindi/libs/indibase/basedevice.h]: This shouldn't be in the header as it breaks all 3rd party drivers not using DSP.

File: libindi/libs/indicom.c [hunk 2]
```diff
@@ -112,7 +113,7 @@ int extractISOTime(const char *timestr, struct ln_date *iso_date)
 
 /* sprint the variable a in sexagesimal format into out[].
  * w is the number of spaces for the whole part.
- * fracbase is the number of pieces a whole is to broken into; valid options:
+ * fracbase is the number of M_PIeces a whole is to broken into; valid options:
```
Comment (knro) [inline libindi/libs/indicom.c]: pi --> M_PI mass replace?

File: libindi/libs/indicom.c [hunk 3]
```diff
@@ -379,7 +380,7 @@ int tty_timeout(int fd, int timeout)
     /* Return -1 due to an error */
     else if (retval == -1)
         return TTY_SELECT_ERROR;
-    /* Return -2 if time expires before anything interesting happens */
+    /* Return -2 if time exM_PIres before anything interesting happens */
```
Comment (knro) [inline libindi/libs/indicom.c]: Same here

File: libindi/libs/indicom.c [hunk 4]
```diff
@@ -721,7 +722,7 @@ int tty_connect(const char *device, int bit_rate, int word_size, int parity, int
     }
 
     // Set raw input (non-canonical) mode, with reads blocking until either a single character
-    // has been received or a one second timeout expires.
+    // has been received or a one second timeout exM_PIres.
```
Comment (knro) [inline libindi/libs/indicom.c]: same

**Comment by iliaplatone** [comment]
-Werror,-Wsign-compare-Werror=maybe-uninitialized causes other parts not to build, is it an error of mine?

**Comment by knro** [comment]
No there are my errors, I'll fix them. I'll merge you PR now, thanks!!

**Pull request merged**